### PR TITLE
Use new option for removing memlock limit

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
Updates the ebpf-manager and uses the new option added in https://github.com/DataDog/ebpf-manager/pull/207 to remove the memlock limit.

### Motivation
The new option does not remove the memlock limit on kernel where eBPF does not use rlimit based memory accounting.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -160,7 +160,7 @@ require (
 	github.com/DataDog/datadog-agent/pkg/util/scrubber v0.59.1
 	github.com/DataDog/datadog-go/v5 v5.6.0
 	github.com/DataDog/datadog-operator v0.7.1-0.20241219210556-f517775059d1
-	github.com/DataDog/ebpf-manager v0.7.4
+	github.com/DataDog/ebpf-manager v0.7.6-0.20241223104659-fb57b17b70e0
```
Comment (guyarb) [inline go.mod:163]: why do we use a commit from ebpf-manager?

Comment (usamasaqib) [reply go.mod:163]: I can cut a new release and use that

**Review by guyarb — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by gjulianm — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]

File: pkg/network/tracer/connection/ebpf_tracer.go [hunk 2]
```diff
@@ -170,16 +169,7 @@ func newEbpfTracer(config *config.Config, _ telemetryComponent.Component) (Trace
...
-		// On some systems, the default for RLIMIT_MEMLOCK may be as low as 64 bytes.
-		// This will result in an EPERM (Operation not permitted) error, when trying to create an eBPF map
-		// using bpf(2) with BPF_MAP_CREATE.
-		//
-		// We are setting the limit to infinity until we have a better handle on the true requirements.
-		RLimit: &unix.Rlimit{
-			Cur: math.MaxUint64,
-			Max: math.MaxUint64,
-		},
+		RemoveRlimit: true,
```
Comment (AmitaiBl) [inline pkg/network/tracer/connection/ebpf_tracer.go:172]: no need for: `MapEditors: make(map[string]*ebpf.Map),`

File: pkg/network/protocols/events/consumer_test.go [hunk 3]
```diff
@@ -189,10 +187,7 @@ func newEBPFProgram(c *config.Config) (*manager.Manager, error) {
 		},
 	}
 	options := manager.Options{
-		RLimit: &unix.Rlimit{
-			Cur: math.MaxUint64,
-			Max: math.MaxUint64,
-		},
+		RemoveRlimit: true,
```
Comment (AmitaiBl) [inline pkg/network/protocols/events/consumer_test.go:190]: no need for : `MapEditors: make(map[string]*ebpf.Map),`?

Comment (usamasaqib) [reply pkg/network/protocols/events/consumer_test.go:190]: There was no need before, so I am preserving that behavior.

File: pkg/network/tracer/connection/ebpf_tracer.go (hunk 2 above)
Comment (usamasaqib) [reply pkg/network/tracer/connection/ebpf_tracer.go:172]: Same as above, I am preserving the old behavior.

**Review by AmitaiBl — APPROVED** [review]

**Pull request merged**